### PR TITLE
Algorithm: Added Min / Max algorithm

### PR DESCRIPTION
**Pull request by Carlos4621**
I have added the Min/Max algorithms, as suggested by the library changes

### CODE REVIEW

File: _Build/Projects/VisualStudio2022/AlgorithmMinMax.h
Comment (AbdullahAmrSobh) [inline _Build/Projects/VisualStudio2022/AlgorithmMinMax.h:1]: I'm not sure, but isn't supposed to located in `SaneCppLibraries\Libraries\Algorithms` ?

Comment (Pagghiu) [reply _Build/Projects/VisualStudio2022/AlgorithmMinMax.h:1]: Yes, it should be in `Libraries\Algorithms`

**Comment by Pagghiu** [comment]
Hi @Carlos4621, thanks for your MR.

The min / max are a little special as they're widely used also in the containers library, so I've been putting them in the `Compiler.h` that is included almost everywhere.

https://github.com/Pagghiu/SaneCppLibraries/blob/2160a852d23d62a5e8c9677bb8297bba3f3d97f6/Libraries/Foundation/Compiler.h#L278

But now that I look at them they're not const T& as they should be, so maybe you could repurpose your MR to fix them.

A few notes: 
- You've been adding the file to the wrong directory as noted in another comment
- Files needs a trailing newline (I will add to coding guidelines)
- Commit messages always use present so it should have been "Algorithms: Add min/max algorithm"

**Comment by Carlos4621** [comment]
Sorry for the confusion, I'm new to contributing projects. Thanks for the recommendations you gave me.

**Pull request closed without merging**